scranShipper/src/RestaurantMenu/index.jsx

```diff
@@ -1,46 +1,78 @@
 import {useEffect, useState} from "react";
-import './style.css'
+import Basket from "../Basket/Basket.jsx";
+
 const RestaurantMenu = (props) => {
+    const [menus, setMenus] = useState([]);
+    const [order, setOrder] = useState({});
+
+    useEffect(() => {
+        fetchMenus(props.restaurantId);
+    }, []);
 
-    console.log(`this is the rest prop from restuarant menu ${props.restaurantId}`);
-    const [menus, getMenu] = useState([]);
     const fetchMenus = (id) => {
-        fetch('https://food-delivery-api.dev.io-academy.uk/restaurants/' + id).then(response => {
-            return response.json();
-        }).then((data) => {
-            getMenu(data.foodItems);
-        })
-    }
+        fetch('https://food-delivery-api.dev.io-academy.uk/restaurants/' + id)
+            .then(response => response.json())
+            .then(data => {
+                setMenus(data.foodItems);
+            });
+    };
+
+    //First checks if foodItem exists in the array and is >= 0 and is valid for subtraction
+    //It then creates a temporary copy of the order array as prevOrder
+    //It then updates the item corresponding to foodItem with the quantity variable
+    const updateOrder = (foodItem, quantity) => {
+        if (!(quantity < 0 && (order[foodItem] === 0 || order[foodItem] === undefined))) {
+            setOrder(prevOrder => ({
+                ...prevOrder,
+                [foodItem]: (prevOrder[foodItem] || 0) + quantity
+            }));
+        }
+    };
 
     useEffect(() => {
         fetchMenus(props.restaurantId)
     }, []);
 
     const displayMenu = (item) => {
```

Comment:
Rather than having a function inside the component, this should be a separate component altogether